src/app/zap-templates/zcl/data-model/chip/service-area-cluster.xml

```diff
@@ -0,0 +1,243 @@
+      <item value="0x10" name="Rubber"/>
+      <item value="0x11" name="Rug"/>
+      <item value="0x12" name="Sand"/>
+      <item value="0x13" name="Stone"/>
+      <item value="0x14" name="Tatami"/>
+      <item value="0x15" name="Terrazzo"/>
+      <item value="0x16" name="Tile"/>
+      <item value="0x17" name="Vinyl"/>
+    </enum>
+
+
+  <bitmap name="Feature" type="bitmap32">
+    <cluster code="0x0150" />
+    <field name="ListOrder"       mask="0x01" />
+  </bitmap>
+
+  <!-- Enum definitions for structures-->
+  <enum name="OpStatusEnum" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Pending"/>
+    <item value="0x01" name="Traversing"/>
+    <item value="0x02" name="Operating"/>
+    <item value="0x03" name="Skipped"/>
+    <item value="0x04" name="Completed"/>
+  </enum>
+
+  <!-- Data types -->
+  <struct name="LocationInfoStruct">
+    <cluster code="0x0150"/>
+    <item name="HomeLocationInfo"  type="HomeLocationStruct"    optional="false" isNullable="true"/>
+    <item name="LandmarkTag"       type="LandmarkTag"           optional="false" isNullable="true"/>
+    <item name="PositionTag"       type="PositionTag"           optional="false" isNullable="true"/>
+    <item name="SurfaceTag"        type="FloorSurfaceTag"       optional="false" isNullable="true"/>
+  </struct>
+
+  <struct name="MapStruct">
+    <cluster code="0x0150"/>
+    <item name="MapId"   type="int8u"                optional="false"/>
+    <item name="Name" type="char_string" length="64" optional="false"/>
+  </struct>
+
+  <struct name="LocStruct">
+    <cluster code="0x0150"/>
+    <item name="LocationId"   type="int32u"                  optional="false" isNullable="false"/>
+    <item name="MapId"        type="int8u"                   optional="false" isNullable="true"/>
+    <item name="LocationInfo" type="LocationInfoStruct"      optional="false" isNullable="false"/>
+  </struct>
+
+  <struct name="ProgStruct">
+    <cluster code="0x0150"/>
+    <item name="LocationId"             type="int32u"        optional="false"/>
+    <item name="Status"                 type="OpStatusEnum"  optional="false"/>
+    <item name="TotalOperationalTime"   type="elapsed_s"     optional="true" isNullable="true"/>
+    <item name="EstimatedTime"          type="elapsed_s"     optional="true" isNullable="true"/>
+  </struct>
+
+
+
+  <!-- Enum definitions for select locations command response-->
+  <enum name="SelectLocationsStatus" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Success"/>
+    <item value="0x01" name="UnsupportedLocation"/>
+    <item value="0x02" name="DuplicatedLocations"/>
+    <item value="0x03" name="InvalidInMode"/>
+    <item value="0x04" name="InvalidSet"/>
+  </enum>
+
+  <!-- Enum definitions for skip location command response-->
+  <enum name="SkipCurrentStatus" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Success"/>
+    <item value="0x01" name="InvalidLocation"/>
```

Comment:
In the spec, the name of this enum is `InvalidLocationList`. Should we keep to the spec?